tmt/steps/report/reportportal.py

```diff
@@ -41,13 +42,34 @@ def _str_env_to_default(option: str, default: Optional[str]) -> Optional[str]:
     return str(os.getenv(env_var))
 
 
-def _filter_invalid_chars(data: str) -> str:
+def _filter_invalid_chars(data: str, step_data: 'ReportReportPortalData') -> str:
     return re.sub(
         '[^\u0020-\uD7FF\u0009\u000A\u000D\uE000-\uFFFD\U00010000-\U0010FFFF]+',
         '',
         data)
 
 
+def _filter_log_per_size(data: str, step_data: 'ReportReportPortalData') -> str:
+    size = len(data)
+    if size > step_data.log_size_limit:
+        return (f"WARNING: Log file was truncated because its size {size} bytes "
+                f"exceeds {step_data.log_size_limit} bytes.\n\n"
+                f"{data[:step_data.log_size_limit]}")
```

Comment:
Wouldn't adding the warning message make the returned str exceed the size limit again?  

```suggestion
                f"{data}"[:step_data.log_size_limit])
```

Reply:
Well, I guess this is a semantics problem. Does the limit apply to the original log or the uploaded log. The primary motivation was the former, though I can definitely modify it.

Reply:
I understood the limit is imposed by RP, therefore it was applied to test logs so they would not be rejected by RP. That would mean the warning needs to be squeezed into the limit as well, trimming the original log a bit more.

Maybe we should keep this filter at the last place, to trim whatever previous filters prepared.

Reply:
The limit is not imposed by RP, it is our decision to introduce the limit as uploading large logs increases RP load and makes its reaction very slow or freeze.. or fail upload eventually. We already encountered few tests producing huge logs (tens of MBs). Moreover, we want to save disk space.

Reply:
I also misunderstood. Make sense then. No objections from my side, except perhaps adding a comment it is not RP imposed limit to prevent future misunderstandings.

Reply:
I see, mea culpa then.

Reply:
I have modified the warning message, please check.

Reply:
Apologies for the late reply. LGTM.